### PR TITLE
Renamed DotNetCoreInstallerV1 to UseDotNetCoreV2 and bumped up version, included nuget installation and proxy setup features.

### PR DESCRIPTION
**Pull request by hiyadav**
This PR does the following changes
1. Renamed DotNetCoreInstallerV1 task to -> UseDotNetCoreV2
2. Bump up the version from 1.0.1 to 2.0.1
3. Implement the functionality to install NuGet (4.4.1) and setup proxy settings (while installing dotnet core sdk/runtime)

### CODE REVIEW

File: Tasks/UseDotNetCoreV2/nugetinstaller.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            console.log(tl.loc("InstallingNuGetVersion", version));
+            await nuGetGetter.getNuGet(version, false, true);
+
+            const proxy: tl.ProxyConfiguration = tl.getHttpProxyConfiguration();
+            if (proxy) {
+                NuGetInstaller.setProxy(proxy);
+            }
+        }
+        catch (error) {
+            tl.error(tl.loc("FailureWhileInstallingNuGetVersion", version, error.message));
```
Comment (sachinma) [inline Tasks/UseDotNetCoreV2/nugetinstaller.ts]: You can just throw from error

File: Tasks/UseDotNetCoreV2/package.json [hunk 2]
```diff
@@ -23,11 +23,14 @@
...
   "homepage": "https://github.com/Microsoft/azure-pipelines-tasks#readme",
   "dependencies": {
+    "@types/mocha": "^5.2.6",
     "@types/node": "^6.0.101",
     "@types/q": "^1.5.0",
-    "vsts-task-lib": "2.0.6",
-    "vsts-task-tool-lib": "0.4.1",
+    "@types/semver": "6.0.0",
+    "packaging-common": "file:../../_build/Tasks/Common/packaging-common-1.0.1.tgz",
+    "semver": "5.6.0",
```
Comment (sachinma) [inline Tasks/UseDotNetCoreV2/package.json]: you should match it with the @types version

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -263,7 +261,7 @@ Tasks/TwineAuthenticateV0/*     @zackrun
 
 Tasks/UniversalPackagesV0/*     @zackrun
 
-Tasks/UseDotNetV2/*     @damccorm
+Tasks/UseDotNetCoreV2/*   @sachinma
```
[the remark this answers is not in the document]
Comment (hiyadav) [reply .github/CODEOWNERS]: I am fine with it. We can rename it to: dotnet. I can do that change on monday, this should not cause any problems in testing. If it blocks you though, please feel free to change it :) 

@pulkitaggarwl FYI, will it be fine to name it: dotnet ?

File: Tasks/UseDotNetV2/task.json [hunk 4]
```diff
@@ -49,7 +49,7 @@
             "type": "string",
             "label": "Version",
             "defaultValue": "2.2.x",
```
Comment (damccorm) [inline Tasks/UseDotNetV2/task.json]: Now that its not required, there should be no defaultValue

**Comment by bryanmacfarlane** [comment]
We can potentially re-open after conversations but the intention is for the installer tasks to continue as the *new* use task stabilizes the pattern.  We shouldn't merge them right now.

**Pull request closed without merging**